hadoop_plugin.py

```diff
@@ -71,11 +64,11 @@ class MetricRecord(object):
     TO_STRING_FORMAT = '[name={},type={},value={},dimensions={}]'
 
     def __init__(self, metric_name, metric_type, value,
-                 dimensions=None, timestamp=None):
+                 dimensions={}):
```

Comment:
This will re-use this dictionary.. if you modify it somewhere in your code then all new callers of this class (that don't specify dimensions) will share the same dictionary object. The old way with =None is the preferred way of doing thing.